examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -415,6 +415,75 @@ function interleaveAttributes( attributes ) {
 
 }
 
+// returns a new, non-interleaved version of the provided attribute
+export function deinterleaveAttribute( attribute ) {
+
+	const cons = attribute.data.array.constructor;
+	const count = attribute.count;
+	const itemSize = attribute.itemSize;
+	const normalized = attribute.normalized;
+
+	const array = new cons( count * itemSize );
+	let newAttribute;
+	if ( attribute.isInstancedInterleavedBufferAttribute ) {
+
+		newAttribute = new InstancedBufferAttribute( array, itemSize, normalized, attribute.meshPerAttribute );
+
+	} else {
+
+		newAttribute = new BufferAttribute( array, itemSize, normalized );
+
+	}
+
+	for ( let i = 0; i < count; i ++ ) {
+
+		newAttribute.setX( i, attribute.getX( i ) );
+
+		if ( itemSize >= 2 ) {
+
+			newAttribute.setY( i, attribute.getY( i ) );
+
+		}
+
+		if ( itemSize >= 3 ) {
+
+			newAttribute.setZ( i, attribute.getZ( i ) );
+
+		}
+
+		if ( itemSize >= 4 ) {
+
+			newAttribute.setW( i, attribute.getW( i ) );
+
+		}
+
+	}
+
+	return newAttribute;
+
+}
+
+// deinterleaves all attributes on the geometry
+export function deinterleaveGeometry( geometry ) {
+
+	for ( const key in geometry.attributes ) {
+
+		if ( geometry.attributes[ key ].isInterleavedBufferAttribute ) {
+
+			geometry.attributes[ key ] = deinterleaveAttribute( geometry.attributes[ key ] );
+
+		}
+
+	}
```

Comment:
Would you mind throwing morph targets in too? 🙏

Reply:
Added morph targets in, as well. Is there ever a case where the attributes object and the morphTargets object could reference the same BufferAttribute object instance?

Reply:
Not in glTF, where the morphs are relative deltas. It wouldn't seem crazy to reuse the attribute for absolute targets (like morph normals or tangents). The more common case would be `uv` and `uv2` sharing the same attribute on the base geometry.

Reply:
> The more common case would be uv and uv2 sharing the same attribute on the base geometry.

Oh good point. I can add a map so any existing reused attributes continued to be structured that way. How would that look? Would they be literally the same BufferAttribute instance (ie `attributes.uv === attributes.uv2`)? Or just the same array / InterleavedBuffer instance?

Reply:
For glTF they'd be the same BufferAttribute instance – I think that's a good enough proxy here too.

Reply:
Great! Done.